tutor/specs/models/courses/offerings/previews.spec.js

```diff
@@ -0,0 +1,139 @@
+import OfferingsMap from '../../../../src/models/course/offerings';
+import CoursesMap from '../../../../src/models/courses-map';
+import Previews from '../../../../src/models/course/offerings/previews';
+
```

Comment:
Great spec! The newish way we make test data is using `Factory.create()` methods that are in `specs/factories/index.js`.  I think you could use:
```js
courses = Factory.coursesMap({ count: 1, should_reuse_preview: true })
offerings = Factory.offeringsMap({ count: 1, is_preview_available: false })
```

That has the advantage of not polluting the global map with test data, since if there's another spec that runs using it you have to be really careful to clear the global map

Reply:
hah, so this got pretty complicated quick due to how the the preview class uses offerings.  I'll take a shot at it later hopefully